### ISSUE TITLE
Throw on RelationalEventId.MigrationsNotFound by default

### ISSUE THREAD

**Issue by AndriySvyryd** [dotnet/efcore#35218]
In EF 9.0 we started throwing for `RelationalEventId.PendingModelChangesWarning`, but in 9.0.1 we've excluded the case when there are no migrations.

**Comment by SamMonoRT** [comment]
Does this need any doc updates? Moving back to 10.0 for tracking.

**Comment by AndriySvyryd** [comment]
@SamMonoRT Yes, if we start throwing on RelationalEventId.MigrationsNotFound we should probably list it in breaking changes
